src/TokenParser/GuardTokenParser.php

```diff
@@ -33,7 +33,11 @@ public function parse(Token $token): Node
 
         $nameToken = $stream->expect(Token::NAME_TYPE);
 
-        $exists = null !== $this->parser->getEnvironment()->$method($nameToken->getValue());
+        try {
+            $exists = null !== $this->parser->getEnvironment()->$method($nameToken->getValue());
+        } catch (\Throwable) {
```

Comment:
should we catch all throwables or only `Twig\Error\Error` ?

Reply:
As there is no contract here (the undefined handlers are not supposed to throw exceptions), I think catching everything is fine, or maybe just `\Exception`?

Reply:
Well, the usage done in Symfony is to throw a SyntaxError, to provide a more precise message than the SyntaxError thrown in https://github.com/twigphp/Twig/blob/91e609663e208dec9b0a70b210d1716a89ca025f/src/ExpressionParser.php#L803 (telling people which extra package they need to install to have access to that function of the twig-bridge).
the `MissingExtensionSuggestor` of `twig/extra-bundle` does exactly the same.

so my suggestion would even be to catch only SyntaxError, to account only for this pattern.

Reply:
If we think Symfony is the only one to mis-behave, I'm fine with catching just Twig exceptions.

Reply:
Done